src/execution_plan/ops/op_unwind.c

```diff
@@ -73,8 +87,18 @@ static OpResult UnwindInit(OpBase *opBase) {
  * NULL will be returned if dynamic list is not evaluted (listIdx = INDEX_NOT_SET)
  * or in case where the current list is fully consumed. */
 Record _handoff(OpUnwind *op) {
-	// If there is a new value ready, return it.
-	if(op->listIdx < SIArray_Length(op->list)) {
+	if(op->is_range && !SIValue_IsNull(op->current)) {
+		if(SIValue_Compare(op->current, op->to, NULL) != 0) {
```

Comment:
This condition is expensive.
Given the start position, end position and step of the range, we can compute the maximum number of `_handoff` calls to perform, `(from - start) / step` e.g. `UNWIND range(0, 100, 2)` can produce up to 50 values. we can compute this number (50) and replace this condition with something along the lines:
```c
if(op->max_iter > 0) {
    op->max_iter--;
    ...
```